src/flows/context/flow.list.tsx

```diff
@@ -48,51 +50,133 @@ export const FlowListContext = React.createContext<FlowListContextType>(
   DEFAULT_CONTEXT
 )
 
+// NOTE: these have no utility, i'm just confused on how we are going to be getting
+// data from the api as a contract hasn't come forward, so i'm trying to be pre-emptive
+// on capabilities to speed integration. Remove the next two functions when that
+// data contract gets some ground and shows up (alex)
+export function serialize(flow) {
+  const apiFlow = {
+    name: flow.name,
+    readOnly: flow.readOnly,
+    pipes: flow.data.allIDs.map(id => {
+      const meta = flow.meta.byID[id]
+
+      return {
+        ...flow.data.byID[id],
+        title: meta.title,
+        visible: meta.visible,
+      }
+    }),
+  }
+
+  return apiFlow
+}
+
+export function hydrate(data) {
+  const flow = {
+    ...JSON.parse(JSON.stringify(EMPTY_NOTEBOOK)),
+    name: data.name,
+    readOnly: data.readOnly,
+  }
+
+  data.pipes.forEach(pipe => {
+    const id = pipe.id || UUID()
+
+    flow.data.allIDs.push(id)
+    flow.meta.allIDs.push(id)
+
+    const meta = {
+      title: pipe.title,
+      visible: pipe.visible,
+      loading: RemoteDataState.NotStarted,
+    }
+
+    delete pipe.title
+    delete pipe.visible
+
+    flow.data.byID[id] = pipe
+    flow.meta.byID[id] = meta
+  })
+
+  return flow
+}
+
 export const FlowListProvider: FC = ({children}) => {
   const [flows, setFlows] = useFlowListState(DEFAULT_CONTEXT.flows)
   const [currentID, setCurrentID] = useFlowCurrentState(null)
 
-  const add = (flow?: Flow): string => {
-    const id = UUID()
+  const add = (flow?: Flow): Promise<string> => {
     let _flow
 
     if (!flow) {
       _flow = {
-        ...EMPTY_NOTEBOOK,
+        ...hydrate({
+          name: 'Name this Flow',
+          readOnly: false,
+          pipes: [
+            {
+              title: 'Select a Metric',
+              visible: true,
+              type: 'queryBuilder',
+              ...JSON.parse(
+                JSON.stringify(PIPE_DEFINITIONS['queryBuilder'].initial)
+              ),
+            },
+            {
+              title: 'Visualize the Result',
+              visible: true,
+              type: 'visualization',
+              ...JSON.parse(
+                JSON.stringify(PIPE_DEFINITIONS['visualization'].initial)
+              ),
+            },
+          ],
+        }),
       }
     } else {
       _flow = {
         name: flow.name,
-        data: flow.data.serialize(),
-        meta: flow.meta.serialize(),
+        data: flow.data,
+        meta: flow.meta,
         readOnly: flow.readOnly,
       }
     }
 
-    setFlows({
-      ...flows,
-      [id]: _flow,
-    })
+    // console.log('add to the api', serialize(data))
```

Comment:
i know i shouldn't leave in commented code. i'm using it here to show where the api plugs into the context mutations, lemme know it there's a better way to communicate this